### ISSUE TITLE
Create abstractions neccessary for plugin system

### ISSUE THREAD

**Issue by winglerw28** [tehgao/mtgdiscordbot#5]
Add a way to treat individual pieces of functionality as isolated components to ensure that future enhancements adhere to SOLID principles and so that bot commands can be easily parameterized.

**Comment by tehgao** [comment]
Agreed -- added a quick and dirty way of doing it via the IHandler interface but there's no guarantee that functionality is mutually exclusive. What ideas did we have?

**Comment by ghost** [comment]
I think the best thing to create an interface at the level of plugins for the bot; rather than having random one-off handlers we could have an `IPlugin` interface that dictates the expected contract between the bot and any additional functionality.

From `IPlugin`, you can have concrete implementations or further interfaces that cascade down from that set of functionality. Event handling, external API communication, etc, could all be contained in its own section and tested on its own. Notably, this would also allow for adding features to the bot without having to change the base functionality of the bot itself, meaning somebody could create a repo just for a plugin if they desired.

**Comment by ghost** [comment]
[My PR](https://github.com/tehgao/mtgdiscordbot/pull/8) should restructure things in a way that adheres to what was discussed above.
